beacon-chain/blockchain/process_block.go

```diff
@@ -282,6 +282,14 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.ReadOnlySignedB
 	if err := s.handleEpochBoundary(ctx, postState, blockRoot[:]); err != nil {
 		return err
 	}
+	// Save state root to block root mapping
+	postStateRoot, err := postState.HashTreeRoot(ctx)
+	if err != nil {
+		return errors.Wrap(err, "could not get postStateRoot")
+	}
```

Comment:
One critical point to consider is how long hash tree root takes here

Reply:
@terencechain I realized the state root is already computed and part of the block when it is proposed, so we can use that instead of computing again. I've updated the PR to reflect this.